demo-simple/src/main/java/com/novoda/downloadmanager/demo/MainActivity.java

```diff
@@ -120,6 +127,15 @@ protected void onCreate(Bundle savedInstanceState) {
 
     private final View.OnClickListener startMigrationOnClick = v -> downloadMigrator.startMigration("downloads.db", migrationCallback);
 
+    private final CompoundButton.OnCheckedChangeListener wifiOnlyOnCheckedChange = (buttonView, isChecked) -> {
+        LiteDownloadManagerCommands downloadManagerCommands = ((DemoApplication) getApplication()).getLiteDownloadManagerCommands();
+        if (isChecked) {
+            downloadManagerCommands.updateAllowedConnectionType(ConnectionType.UNMETERED);
```

Comment:
Should we be handling the case where the client passes us `Null`? it will crash internally if they do right.

Reply:
I think most other methods on the DLM will have a guard against nulls either when retrieving from a map or checking against other objects.

Reply:
Good point... it's however right to crash if that happens, to me. What about if we throw an `IllegalArgumentException` in that case?

Reply:
In this case it is probably valid to throw a `NullPointer` now that you mention it, the others work on ids so if you get an invalid id you just swallow the problem. Here if you pass an incorrect connection type it is strange, especially as it is normally an enum 😄

Reply:
https://github.com/novoda/download-manager/pull/318/commits/2d03303065f9c816e8e46e455c4bdb48ed14f218